controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -1519,6 +1532,27 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
           targetNamespaceDisplayName: subgraph.namespace,
         });
 
+        if (
+          opts.openaiApiKey &&
+          // Avoid calling OpenAI API if the schema is too big.
+          // Best effort approach. This way of counting tokens is not accurate.
+          subgraph.schemaSDL.length <= 10_000
```

Comment:
You can count tokens e.g. using a library like this:
https://github.com/dqbd/tiktoken

Reply:
Thanks, I saw it but I'm a bit reluctant to run WASM in our hot paths. It requires more testing.